services/src/main/scala/cromwell/services/metadata/impl/archiver/ArchiveMetadataSchedulerActor.scala

```diff
@@ -86,22 +86,31 @@ class ArchiveMetadataSchedulerActor(archiveMetadataConfig: ArchiveMetadataConfig
       // paths are actively receiving data points throughout:
-      archiveNextWorkflows(startTime).onComplete({
-        case Success(cnt) if cnt > 0 =>
-          1 to cnt.toInt foreach { _ =>
-            increment(workflowsProcessedSuccessMetricPath, ServicesPrefix)
-            count(workflowsProcessedFailureMetricPath, 0L, ServicesPrefix)
+      archiveNextWorkflows.onComplete({
+        case Success(cnt) =>
+          count(workflowsProcessedSuccessMetricPath, cnt, ServicesPrefix)
+          count(workflowsProcessedFailureMetricPath, 0L, ServicesPrefix)
+
+          if (cnt > 0) {
+            if (archiveMetadataConfig.debugLogging) log.info(s"Successfully archived $cnt workflows.")
+            self ! ArchiveNextWorkflowMessage
+          } else {
+            count(rowsProcessedMetricPath, 0L, ServicesPrefix)
+            sendGauge(workflowsToArchiveMetricPath, 0L, ServicesPrefix)
+            sendTiming(workflowArchiveTotalTimeMetricPath, calculateTimeSince(startTime), ServicesPrefix)
+            if (archiveMetadataConfig.debugLogging) log.info(s"No complete workflows which finished over ${archiveMetadataConfig.archiveDelay} ago remain to be archived. Scheduling next poll in ${archiveMetadataConfig.backoffInterval}.")
+            scheduleNextWorkflowToArchive()
           }
-          sendTiming(workflowArchiveTotalTimeMetricPath, calculateTimeSince(startTime) / cnt, ServicesPrefix)
-          self ! ArchiveNextWorkflowMessage
+        case Success(cnt) if cnt > 0 =>
+//          count(workflowsProcessedSuccessMetricPath, cnt, ServicesPrefix)
+//          count(workflowsProcessedFailureMetricPath, 0L, ServicesPrefix)
         case Success(_) =>
-          count(rowsProcessedMetricPath, 0L, ServicesPrefix)
-          count(workflowsProcessedSuccessMetricPath, 0L, ServicesPrefix)
-          count(workflowsProcessedFailureMetricPath, 0L, ServicesPrefix)
-          sendGauge(workflowsToArchiveMetricPath, 0L, ServicesPrefix)
-          sendTiming(workflowArchiveTotalTimeMetricPath, calculateTimeSince(startTime), ServicesPrefix)
-          scheduleNextWorkflowToArchive()
-          if (archiveMetadataConfig.debugLogging) log.info(s"No complete workflows which finished over ${archiveMetadataConfig.archiveDelay} ago remain to be archived. Scheduling next poll in ${archiveMetadataConfig.backoffInterval}.")
+//          count(rowsProcessedMetricPath, 0L, ServicesPrefix)
+//          count(workflowsProcessedSuccessMetricPath, 0L, ServicesPrefix)
+//          count(workflowsProcessedFailureMetricPath, 0L, ServicesPrefix)
+//          sendTiming(workflowArchiveTotalTimeMetricPath, calculateTimeSince(startTime), ServicesPrefix)
+//          scheduleNextWorkflowToArchive()
+//          if (archiveMetadataConfig.debugLogging) log.info(s"No complete workflows which finished over ${archiveMetadataConfig.archiveDelay} ago remain to be archived. Scheduling next poll in ${archiveMetadataConfig.backoffInterval}.")
         case Failure(error) =>
```

Comment:
In the case of failure I think `archiveSummaryEntries` / `archiveNextWorkflows` would simply return a single `Future.failed` describing the problem and no additional `Long` indicating how many successes there were in addition.

How do you feel about counting the entire batch as failed? The metrics should be reasonably meaningful on non-golden paths, but I'm not seeing a good way to make them exactly accurate.
```
count(workflowsProcessedFailureMetricPath, archiveMetadataConfig.batchSize, ServicesPrefix)
```

Reply:
Sounds reasonable to me. I think we may end up over-estimating how bad failure situations are (it will look like nothing is being done when actually 19/20 workflows may be being processed successfully), but in terms of us being alerted that's probably preferable to under-reporting.

The "fix" would be to either report back the archive results as a `Seq[Try[]]`s - one per archive attempt, or move _all_ the metrics into the per-archive code. I think what we have now is a reasonable "good enough" answer for now